### ISSUE TITLE
False positive Unused import of a record class when namespace require uses ':refer :all`

### ISSUE THREAD

**Issue by mrkam2** [clj-kondo/clj-kondo#2223]
`"clj-kondo-version": "2023.10.21-SNAPSHOT"`

**problem**

False positive on record class import when its namespace is required with `:refer :all`.

**repro**

`a.clj`:
```clj
(ns a)

(defrecord A [a])
```

`b.clj`:
```clj
(ns b
  (:require
    [a :refer :all])
  (:import
    (a A)))

(defn tst [^A x]
  (.-a x))
```

```
$ clj-kondo --lint src       
src/b.clj:5:8: warning: Unused import A
linting took 116ms, errors: 0, warnings: 1
```

**expected behavior**

No warning.

**Comment by mrkam2** [comment]
If I drop the import, I get 
```
; Syntax error (IllegalArgumentException) compiling fn* at (src/b.clj:5:1).
; Unable to resolve classname: A
; Evaluation of file b.clj failed: class clojure.lang.Compiler$CompilerException
```

If I use require with an alias, there is no warning, for the following `ns` declaration:
```clj
(ns b
  (:require
    [a :as a])
  (:import
    (a A)))
```

**Comment by borkdude** [comment]
Thanks!

**Comment by borkdude** [comment]
I'm fixing this. The reason why not many people have run into this, is that the above is often considered too low level. You should not be using interop to access record stuff, neither should you need to use type hints. Record fields can simply be accessed using keywords. A more idiomatic style:

```
(defn tst [x]
  (:a x))
```
